src/test/java/com/openlattice/rehearsal/datasets/DataSetsTest.kt

```diff
@@ -0,0 +1,107 @@
+package com.openlattice.rehearsal.datasets
+
+import com.openlattice.organization.OrganizationIntegrationAccount
+import com.openlattice.organizations.Organization
+import com.openlattice.rehearsal.authentication.MultipleAuthenticatedUsersBase
+import junit.framework.Assert
+import org.junit.BeforeClass
+import org.junit.Test
+import java.sql.DriverManager
+import java.sql.SQLException
+import java.util.*
+
+
+open class DataSetsTest : MultipleAuthenticatedUsersBase() {
+    companion object {
+        lateinit var credentials: OrganizationIntegrationAccount
+        lateinit var organization: Organization
+        lateinit var organizationID: UUID
+        lateinit var url: String
+
+        const val create_query = "CREATE TABLE IF NOT EXISTS " +
+                "authors (id serial PRIMARY KEY, name VARCHAR(25));"  +
+                "INSERT INTO authors(id, name) VALUES(1, 'Jack London');"
+        val delete_column_query = "ALTER TABLE authors " +
+                "DROP COLUMN name;"
+        val delete_table_query = "DROP TABLE authors;"
+        val tableName = "authors"
+        val columnName = "name"
+
+        @JvmStatic
+        @BeforeClass
+        fun init() {
+            loginAs("admin")
+
+            organization = createOrganization()
+            organizationID = organization.id
+//            organizationID = UUID(1L, 0L)
+            url = "jdbc:postgresql://localhost:5432/org_" + organizationID.toString().replace("-", "")
+            credentials = organizationsApi.getOrganizationIntegrationAccount(organizationID)
+       }
+
+    }
+
+    fun uploadData() {
+        val id = 1
+        val author = "Jack London"
+
+        try {
+            DriverManager.getConnection(url, credentials.user, credentials.credential).use { con ->
+                con.prepareStatement(create_query).executeUpdate()
+            }
+        } catch (ex: SQLException) {
+            logger.error("Couldn't create table.", ex)
+        }
+    }
+
+    fun removeDataColumn() {
+        try {
+            DriverManager.getConnection(url, credentials.user, credentials.credential).use { con ->
+                con.prepareStatement(delete_column_query).executeUpdate()
+            }
+        } catch (ex: SQLException) {
+            logger.error("Couldn't create table.", ex)
+        }
+    }
+
+    fun removeDataTable() {
+        try {
+            DriverManager.getConnection(url, credentials.user, credentials.credential).use { con ->
+                con.prepareStatement(delete_table_query).executeUpdate()
+            }
+        } catch (ex: SQLException) {
+            logger.error("Couldn't create table.", ex)
+        }
+    }
+
+    @Test
+    fun getMetadata() {
+        uploadData()
+        Thread.sleep(30000L)
```

Comment:
is 30 seconds the frequency the background job runs at? 

Including Thread.sleep() calls like is gonna result in rehearsal taking suuuuper long to run, so we should avoid them / shorten them wherever possible. If the reason for it is that we're waiting on a background job to execute, I'd prefer switching to a retry loop that executes every few seconds and exits/fails after 30 seconds.

Reply:
(and same for the Thread.sleeps below)